src/robot.coffee

```diff
@@ -247,6 +264,28 @@ class Robot
   # Returns nothing.
   # Returns before executing callback
   receive: (message, cb) ->
+    # When everything is finished (down the middleware stack and back up),
+    # pass control back to the robot
+    receiveMiddlewareDone = () ->
+      if cb?
+        process.nextTick -> cb true
+
+    @middleware.receive.execute(
+      {response: new Response(this, message)}
```

Comment:
If we _don't_ have to access to a `response` object, then in order to send a message to the user/room that initiated, we'd need to do something like:

```coffeescript
envelope =
  room: context.message.room
  user: context.message.user
  message: context.message
robot.adapter.send envelop, "you've been denied"
```

The use case is having a receive middleware that blacklists specific users from using _any_ commands. So, it checks for the robot.respond regex, and replies they aren't trusted.